tests/test_service.py

```diff
@@ -247,171 +247,74 @@ def test_update_recommendation_between_products(self):
+        
+        # Test invalid related_product_id
         invalid_recommendation = {
             "product-id": old_recommendation.id,
-            "related-product-id": "Wrong_id",
+            "related-product-id": "invalid_related_product_id",
             "type-id": new_typeid[old_recommendation.typeid],
             "status": True
         }
 
-        resp = self.app.put("/recommendations",
-                            json=invalid_recommendation,
-                            content_type="application/json")
+        resp = self.app.put(update_url, json=invalid_recommendation, content_type="application/json")
         self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
 
-        resp = self.app.get\
-            ("/recommendations/relationship",
-             query_string=dict(product1=old_recommendation.id,
-                               product2=old_recommendation.rel_id)
-             )
-        updated_recommendation = Recommendation()
-        updated_recommendation.deserialize(resp.get_json())
-
-        self.assertEqual(updated_recommendation,
-                         old_recommendation,
-                         "recommendation should not be updated")
-
+        # Test invalid type
         invalid_recommendation = {
             "product-id": old_recommendation.id,
             "related-product-id": old_recommendation.rel_id,
             "type-id": 10,
             "status": True
         }
 
-        resp = self.app.put("/recommendations",
-                            json=invalid_recommendation,
-                            content_type="application/json")
+        resp = self.app.put(update_url, json=invalid_recommendation, content_type="application/json")
         self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
 
-        resp = self.app.get\
-            ("/recommendations/relationship",
-             query_string=dict(product1=old_recommendation.id,
-                               product2=old_recommendation.rel_id)
-             )
-        updated_recommendation = Recommendation()
-        updated_recommendation.deserialize(resp.get_json())
-
-        self.assertEqual(updated_recommendation,
-                         old_recommendation,
-                         "recommendation should not be updated")
-
-        does_not_exist_recommendation = {
+        # Test non-existe product_id
+        non_exist_recommendation = {
             "product-id": 50000,
             "related-product-id": old_recommendation.rel_id,
             "type-id": 2,
             "status": True
         }
 
-        resp = self.app.put("/recommendations",
-                            json=does_not_exist_recommendation,
-                            content_type="application/json")
+        resp = self.app.put(update_url, json=non_exist_recommendation, content_type="application/json")
         self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
 
-        resp = self.app.get\
```

Comment:
Don't we need to verify if the recommendation has been actually updated in the database?

Reply:
I think on tests/test_service.py, line 284-287 does that.

Reply:
Oh okay